zipkin-server/src/main/java/zipkin/server/ActuatorConditionalImporter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2015-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.server;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+/**
+ * Actuator relies on classpath scanning, so we cannot choose one configuration entrypoint unless
```

Comment:
Nitpick. There is no classpath scanning involved in autoconfiguration. I would like to see the wording changed because enough people are freaked out by the concept of classpath scanning, and we wouldn't like inaccurate descriptions to get attached. I'm not sure what to suggest because I don't know what the intent of this paragraph is really. Is it just autoconfiguration that is disabled and relied on?

Reply:
I think it does rely on it, but it could be changed to not. For example `BeansEndpointAutoConfiguration` is only found through META-INF spring factories, which implies scanning, right?

Reply:
No. Not at all. The use of `spring.factories` is more or less entirely to prevent classpath scanning. You don't need to scan if you have a list of classes to import via a properties file.

Reply:
by changed not to, I mean actuator could have an entrypoint type that includes imports of all the autoconfiguration it manages instead of having each loaded by meta-inf. As each of the endpoints is already subject to secondary config based on enabled properties this could work. Ex this would change this to a much more manageable thing where we look for a class name once vs N* endpoints defined. OTOH, I don't think upstream probably cares to optimize for this.

Meanwhile it is important to keep actuator optional, and this is basically an alternative to looking up the spring factories that have package name prefix related to actuator.

Reply:
I'll remove the comment about "spring.factories"

meanwhile we get constant complaints about startup time. team members here's teams complain, windows starts in 10s and I noticed as you mentioned actuator takes a notable percentage off the startup. We can pay attention to what we want but meanwhile some mass of effort is spent, maybe not on classpath scanning, but whatever that effort is, less of it is occurring now.

Reply:
Boot 2.2 enables only 2 actuators by default. It's much faster. Did you fold that in yet? Users would have to explicitly enable all the others that they want to expose. Seems like a legitimate task for a Zipkin server operator.

Reply:
we haven't folded 2.2 in.

Custom servers are unsupported and we define custom as anything not defined in env variables. I would consider expectations around autoconfiguration tuning in the realm of that unsupported area though folks obviously can self support that. In other words, our operations are not defined as "a usual spring boot app" as that implies a support load we cannot afford.